kiosk/src/Components/ScanQR.tsx

```diff
@@ -26,11 +26,16 @@ const ScanQR: React.FC<IProps> = ({ kiosk }) => {
     const checkUrl = async () => {
         const input = document.getElementById("kiosk-share-link") as HTMLInputElement;
         const inputValue = input.value;
-        const shareLink = /(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S\d{5}-\d{5}-\d{5}-\d{5})|(?:_[a-zA-Z0-9]+))/.exec(inputValue);
+        const shareLink = /^(https:\/\/)((arcade\.makecode\.com\/)|(makecode\.com\/))((?:S\d{5}-\d{5}-\d{5}-\d{5})$|(?:_[a-zA-Z0-9]+)$)/.exec(inputValue);
+        const shareCode = /(^(?:S\d{5}-\d{5}-\d{5}-\d{5})$|^(?:_[a-zA-Z0-9]{12})$)/.exec(inputValue);
         let shareId;
         if (shareLink) {
-            setLinkError(false);
             shareId = /\/([^\/]+)\/?$/.exec(inputValue)?.[1];
+        } else if (shareCode) {
+            shareId = shareCode?.[1];
```

Comment:
```suggestion
            shareId = shareCode[1];
```

if we know `shareCode` is non-null `?.` won't do anything